### PR TITLE
Rework axis aligned bounding box

### PR DESCRIPTION
**Pull request by TomFischer**
Until now, it was always necessary to enlarge the axis aligned bounding box (AABB) to assure that all points of the point set that defined the AABB are located inside this AABB. AABB should implement a from the right half-open interval. Unfortunately the implementation was not consistent.

The PR changes this behaviour. The AABB will enlarged a very small amount (unit in the last place) by default. The method `containsPoint()` is implemented such that "min <= p < max".

The doxygen documentation for the class template is not generated. If somebody has a hint ...

ToDo
 - [x] fix documentation

### CODE REVIEW

File: GeoLib/AABB.h [hunk 1]
```diff
@@ -160,9 +181,33 @@ class AABB
 	{
 		init(*pnt);
 	}
+
+	/// Updates the bounding box using point \f$p\f$ without resizing it.
+	std::pair<bool,std::bitset<3>> update_(PNT_TYPE const & p)
```
Comment (endJunction) [inline GeoLib/AABB.h]: Would you please elaborate more on what update without resizing means?
Maybe return value explanation would already help.

File: GeoLib/AABB.h [hunk 2]
```diff
@@ -150,6 +160,17 @@ class AABB
 		std::numeric_limits<double>::lowest(),
 		std::numeric_limits<double>::lowest()}}};
 private:
+	/// Enlarge the bounding box the smallest possible amount.
+	void enlarge(std::bitset<3> to_update = 7)
```
Comment (endJunction) [inline GeoLib/AABB.h]: Please add a note that only right boundary is moved.

**Comment by endJunction** [comment]
What is the difference btw. `update` and `update_`? Maybe there is a better name?

**Comment by TomFischer** [comment]
Renamed `update_()` to `updateWithoutEnlarge()`.

**Comment by endJunction** [comment]
:fast_forward:

**Comment by TomFischer** [comment]
All changes included in PR #929.

**Pull request closed without merging**